starters/e2e/qwikcity/page.spec.ts

```diff
@@ -108,6 +108,45 @@ function tests() {
         h1: "Product: tshirt",
         activeHeaderLink: "Products",
       });
+
+      /***********  Products: shirt (rewrite to /products/tshirt)  ***********/
+      await linkNavigate(
+        ctx,
+        '[data-test-link="products-shirt-rewrite-plain_string"]',
+      );
+      await assertPage(ctx, {
+        pathname: "/qwikcity-test/products/shirt-rewrite-plain_string/",
+        title: "Product tshirt - Qwik",
+        layoutHierarchy: ["root"],
+        h1: "Product: tshirt",
+        activeHeaderLink: "Products",
+      });
+
+      /***********  Products: shirt (rewrite to /products/tshirt)  ***********/
+      await linkNavigate(
+        ctx,
+        '[data-test-link="products-shirt-rewrite-url_object"]',
+      );
+      await assertPage(ctx, {
+        pathname: "/qwikcity-test/products/shirt-rewrite-url_object/",
+        title: "Product tshirt - Qwik",
+        layoutHierarchy: ["root"],
+        h1: "Product: tshirt",
+        activeHeaderLink: "Products",
+      });
+
+      /***********  Products: shirt (rewrite to /products/tshirt)  ***********/
+      await linkNavigate(
+        ctx,
+        '[data-test-link="products-shirt-rewrite-plain_string_no_trailing_slash"]',
+      );
+      await assertPage(ctx, {
```

Comment:
also add a test for clearing and adding search parameters

Reply:
I added a test with search params, making sure they are properly present after a rewrite